pallets/ah-migrator/src/benchmarking.rs

```diff
@@ -0,0 +1,105 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! To run these benchmarks, you will need a modified version of `frame-omni-bencher` that can load
+//! snapshots of the relay and asset hub. You can find it on branch `oty-ahm-omni-bencher` of the
+//! SDK. Install it with
+//! `cargo install --path substrate/utils/frame/omni-bencher --profile production`
+//!
+//! ```bash
+//! frame-omni-bencher v1 benchmark pallet --runtime=target/release/wbuild/asset-hub-polkadot-runtime/asset_hub_polkadot_runtime.wasm --pallet "pallet-ah-migrator" --extrinsic "" --snap=ah-polkadot.snap --rc-snap=polkadot.snap
+//! ```
+
+use crate::*;
+use core::str::FromStr;
+use cumulus_primitives_core::{AggregateMessageOrigin, InboundDownwardMessage};
+use frame_benchmarking::v2::*;
+use frame_support::{traits::EnqueueMessage, weights::WeightMeter};
+use frame_system::RawOrigin;
+use pallet_rc_migrator::types::PalletMigration;
+use xcm::VersionedXcm;
+
+#[benchmarks(where T: pallet_balances::Config)]
+mod benchmarks {
+	use super::*;
+
+	#[benchmark]
+	fn receive_multisigs() {
+		verify_snapshot::<T>();
+		let (messages, _cursor) = relay_snapshot(|| {
+			unwrap_no_debug(pallet_rc_migrator::multisig::MultisigMigrator::<T>::migrate_out_many(
```

Comment:
We possibly need an alternative limit here, like 100 or so. Otherwise if we only pass the weight then we dont know how many it will be before hand.